readthedocs/projects/forms.py

```diff
@@ -202,6 +201,22 @@ class Meta(object):
             # 'num_major', 'num_minor', 'num_point',
         )
 
+    def __init__(self, *args, **kwargs):
+        super(ProjectAdvancedForm, self).__init__(*args, **kwargs)
+
+        default_choice = (None, '-' * 9)
+        versions = self.instance.versions.values_list('slug', 'verbose_name')
+        self.fields['default_branch'].widget = forms.Select(
+            choices=[default_choice] + list(versions)
```

Comment:
For default branch we need to list ALL versions (branches and tags, active and inactive ones)